### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by MPrzem**
Pull request with done homework

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 -Werror -Iinc src/*.cpp -o main
```
Comment (Marcinati) [inline Makefile]: You can add more compile flag, like:  ```-Wall -Wextra -pedantic -Wconversion -O3```

File: src/main.cpp [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #include <vector>
 #include <iostream>
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
 #include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
```
Comment (Marcinati) [inline src/main.cpp]: You have included functions in previous line, you can delete this line and comment above.

File: tutor [hunk 3]
```diff
@@ -0,0 +1,974 @@
...
+  It is a good book to get to know almost anything you want to do with Vi.
+  The sixth edition also includes information on Vim.
+
+  This tutorial was written by Michael C. Pierce and Robert K. Ware,
+  Colorado School of Mines using ideas supplied by Charles Smith,
+  Colorado State University.  E-mail: bware@mines.colorado.edu.
+
+  Modified for Vim by Bram Moolenaar.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Marcinati) [inline tutor:971]: Ok. :)

File: src/functions.cpp [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
-#include <algorithm>
 #include <iostream>
+#include <algorithm>
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
 #include <limits>
```
Comment (Marcinati) [inline src/functions.cpp:4]: Please, sort headers and remove comment.

File: src/functions.cpp [hunk 5]
```diff
@@ -1,6 +1,6 @@
...
 
 void print(std::vector<int> const & numbers) {
```
Comment (Marcinati) [inline src/functions.cpp:6]: Is this for loop work with int i? I'm strongly recommend to use size_t instead of int.

**Review by Marcinati — CHANGES_REQUESTED** [review]
If you will add more compile flags and add some improvements it'll be allright.

**Review by DamiXioR — COMMENTED** [review]
If you add more compile flags you will be alerted about use std::size_t or unsigned int instead int.

File: tutor [hunk 6]
```diff
@@ -0,0 +1,974 @@
...
+  3. Now move to the fourth line.
+  4. Type   2dd   to delete two lines.
+
+--->  1)  Roses are red,
+--->  2)  Mud is fun,
+--->  3)  Violets are blue,
+--->  4)  I have a car,
+--->  5)  Clocks tell time,
+--->  6)  Sugar is sweet
+--->  7)  And so are you.
```
Comment (ziobron) [inline tutor]: not completed

File: tutor [hunk 7]
```diff
@@ -0,0 +1,974 @@
...
+     first error.
+  2. Type  x  to delete the first unwanted character.
+  3. Now type  u  to undo the last command executed.
+  4. This time fix all the errors on the line using the  x  command.
+  5. Now type a capital  U  to return the line to its original state.
+  6. Now type  u  a few times to undo the  U  and preceding commands.
+  7. Now type CTRL-R (keeping CTRL key pressed while hitting R) a few times
+     to redo the commands (undo the undo's).
+
+---> Fiix the errors oon thhis line and reeplace themRRRtth undo.
```
Comment (ziobron) [inline tutor]: not completed

File: tutor [hunk 8]
```diff
@@ -0,0 +1,974 @@
...
+  2. Place the cursor on the  u  in  lubw.
+
+  3. Type  ce  and the correct word (in this case, type  ine ).
+
+  4. Press <ESC> and move to the next character that needs to be changed.
+
+  5. Repeat steps 3 and 4 until the first sentence is the same as the second.
+
+---> This line has a few words that need changing usf the change operator.
+---> This line has a few words that need changing using the change operator.
```
Comment (ziobron) [inline tutor]: not completed

File: tutor [hunk 9]
```diff
@@ -0,0 +1,974 @@
...
+  2. The motions are the same, such as   w (word) and  $ (end of line).
+
+  3. Move the cursor to the first line below marked --->.
+
+  4. Move the cursor to the first error.
+
+  5. Type  c$  and type the rest of the line like the second and press <ESC>.
+
+---> The end of this line needs some help to make it like the second.
+---> The end of this line needs to be corrected using the  c$  command.
```
Comment (ziobron) [inline tutor]: not completed

File: tutor [hunk 10]
```diff
@@ -0,0 +1,974 @@
...
+
+  1. Move the cursor to the line below marked --->.
+
+  2. Type  :s/thee/the <ENTER>  .  Note that this command only changes the
+     first occurrence of "thee" in the line.
+
+  3. Now type  :s/thee/the/g .  Adding the  g  flag means to substitute
+     globally in the line, change all occurrences of "thee" in the line.
+
+---> thee best time to see thee flowers is in thee spring.
```
Comment (ziobron) [inline tutor]: not completed

**Review by ziobron — CHANGES_REQUESTED** [review]
Not everything is completed in vimtutor. Also there are some compilation flags missing `-Wall -Wextra -pedantic -Wconversion -O3` as Marcinati suggested. Please deliver them :)

**Comment by MPrzem** [comment]
> You left comment. ;)

Thanks for review. Could you tell me where i've left comment?

File: src/main.cpp [hunk 11]
```diff
@@ -1,7 +1,6 @@
 #include <vector>
 #include <iostream>
 #include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
```
Comment (ziobron) [inline src/main.cpp:3]: Here is a // TODO comment. You can/should remove it when this job is done :)

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
+4 XP Narzędzia#2 – vimtutor
+5 XP Narzędzia#2 – git - konflikty
+4 XP Narzędzia#2 – flagi kompilacji
+3 XP Narzędzia#2 – git - cherrypick

{ "requirement_ids": [1999, 2000, 2001, 2002], "points": [4, 5, 4, 3] }

**Pull request closed without merging**